policy/testdata/README.md

```diff
@@ -0,0 +1,170 @@
+# CEL Policy
+
+The Common Expression Language (CEL) supports simple expressions: no variables, functions, or modules. However, CEL expression graphs can be composed together, allowing for reuse and development clarity which is not otherwise possible within CEL.
+
+To address this case, we're introducing the CEL Policy format which is fully runtime compatible with CEL. All of the same performance and safety hardening guarantees which apply to CEL also apply to CEL Policy. The net effect is significantly improved authoring and testability. The YAML-based policy format is easily extensible and inspired by [Kubernetes Admission Policy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/) with CEL.
+
+## Policy Language
+
+A policy is a named instance of a rule which consists of a set of conditional outputs and conditional sub-rules. Matches within the rule and subrules are combined and ordered according to the policy evaluation semantic. The default semantic is `FIRST_MATCH`. The supported top-level fields in a policy include: `name`, `description`, `imports`, and `rule.`
+
+### Rule
+
+The `rule` node in a policy is the primary entry point to CEL computations. Fields above the `rule` are intended to simplify or support the CEL expressions within the `rule` block. For example, the [`imports` list refers to a set of type names](#imports) who should be imported by their simple name within the CEL expressions contained in the `rule`.
+
+#### Variables
+
+A \`rule\` has a single `variables` block. Variables are written as an ordered list. Variables may refer to another variable; however, the  variable must be declared before use, i.e. be defined before it is referenced. A variable has a `name` and an `expression`.
+
+```
+variables:
+  - name: first_item
+    expression: "1"
+  - name: list_of_items
+    expression: "[variables.first_item, 2, 3, 4]"
+```
+
+Variables in CEL Policy are lazily evaluated and memoized as CEL is side-effect free. Only the variables which are accessed during a `match` `condition` or an `output` are evaluated. The use of a variable is equivalent to using the `cel.bind()` macro to introduce local computations within a CEL expression.
+
+#### Match
+
+A `rule` has a single `match` block. The match block should have at least one `output` value, though `output` expressions may be `condition`al. The default evaluation order for the sequence of matches is top-down, first-match.
+
+```
+rule:
+  match:
+    - condition: "request.user.name.startsWith('j')"
+      output: "Hi, J!"
+    - output: "Hi, " + request.user.name + "!"
```

Comment:
If the `condition` is missing, what does it mean?

Reply:
I updated the text right after this example. The `condition` defaults to `true` if not specified.